src/API.jl

```diff
@@ -348,7 +348,18 @@ function driverconnect(connstr)
     return dbc
 end
 
-connect(dsn, user, pwd) = driverconnect(string("DSN=", dsn, user === nothing ? "" : ";UID=$user", pwd === nothing ? "" : ";PWD=$pwd"))
+connect(dsn, extraauth) = driverconnect("$dsn;$extraauth")
+
+# should this be somewhere else? utils?
+# also question on escaping: apparently e.g. PWD={$pwd} is the correct way to escape a parameter with semicolons in it.
+# however there seems no standard way to escape PWD=mypass}hasacurly. (SQL server wants double }}, but other drivers differ).
+function getextraauth(usr::Union{AbstractString, Nothing}, pwd::Union{AbstractString, Nothing}, extraauth::Union{AbstractString, Nothing}) :: String
+    return join([
+        usr === nothing ? "" : "UID={$usr}",
```

Comment:
This is more escaping that we had before! So I say we run w/ this for now; the nice thing is that people can always just pass the whole connection string themselves if they want.

Reply:
woop. I did have one thought on this - my change is *technically* a breaking change for any user who is currently passing their password manually like `connect(dsn, "root", "{im escaping my pwd myself}")`. Do you have any thoughts on that? it's a bit of an edge case and I can't see any issues discussing it so maybe no-one has hit it yet.